File: UI/pc-part-picker/src/model/Pccase.ts
import { FormfactorType } from "./FormfactorType";
import { ManufacturerType } from "./ManufacturerType";
import { PcPart } from "./Pcpart";

export interface Pccase extends PcPart{
    formFactor: string | null;
    color: string | null;
    sidePanel: string | null;
    externalVolume: number | null;
    internal35Bays: number | null;
    formFactorTypeId: number | null;
    manufacturerTypeId: number | null;
    formFactorType: string | null;
    manufacturerType: string | null;
}

File: API/PCPartPicker/Models/Pccase.cs
using PCPartPicker.Interfaces;
using System;
using System.Collections.Generic;

namespace PCPartPicker.Models;

public partial class Pccase : IComponent
{
    public int Id { get; set; }

    public string? Name { get; set; }

    public decimal? Price { get; set; }

    public string? FormFactor { get; set; }

    public string? Color { get; set; }

    public string? SidePanel { get; set; }

    public decimal? ExternalVolume { get; set; }

    public int? Internal35Bays { get; set; }

    public int? FormFactorTypeId { get; set; }

    public int? ManufacturerTypeId { get; set; }

    public virtual ICollection<Configuration> Configurations { get; set; } = new List<Configuration>();

    public virtual FormfactorType? FormFactorType { get; set; }

    public virtual ManufacturerType? ManufacturerType { get; set; }
}
